cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -341,24 +341,31 @@ private CompletionStage<Map<String, AvailabilityHint>> createSnapshot(RingRespon
                     LOGGER.info("Creating snapshot on instance snapshotName={} keyspace={} table={} datacenter={} fqdn={}",
                                 snapshotName, maybeQuotedKeyspace, maybeQuotedTable, datacenter, ringEntry.fqdn());
                     SidecarInstance sidecarInstance = new SidecarInstanceImpl(ringEntry.fqdn(), sidecarClientConfig.effectivePort());
-                    createSnapshotFuture = sidecar
-                                           .createSnapshot(sidecarInstance, maybeQuotedKeyspace, maybeQuotedTable, snapshotName)
-                                           .handle((resp, throwable) -> {
-                                               if (throwable == null)
-                                               {
-                                                   // Create snapshot succeeded
-                                                   return hint;
-                                               }
-
-                                               if (isExhausted(throwable))
-                                               {
-                                                   LOGGER.warn("Failed to create snapshot on instance", throwable);
-                                                   return PartitionedDataLayer.AvailabilityHint.DOWN;
-                                               }
-
-                                               LOGGER.error("Unexpected error creating snapshot on instance", throwable);
-                                               return PartitionedDataLayer.AvailabilityHint.UNKNOWN;
-                                           });
+                    String resolvedSnapshotTtl = options.clearSnapshot() ? options.effectiveSnapshotTtl() : null;
```

Comment:
The condition is reversed. The resolved snapshot ttl is set, when `!options.clearSnapshot()`.

Reply:
I think the condition here is correct. When clear snapshot is set, it allows the snapshot TTL to be set. When clear snapshot is set to false, we don't want the TTL to clear the snapshot at a later point. I think that's how I understand it

Reply:
my understanding is when clearSnapshot is true, the snapshot is cleared on job completion; otherwise it sets a TTL and let Cassandra server to delete. 
I'd wait for Saranya to clarify.